enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,1202 @@
+customExtensionSpecs.AddLabel(commonlabels.SLOW) 
+
+// similar story with tags.
+customExtensionSpecs.AddTag({"mytag": "myvalue"})  
+
+// Here, OpenShift convention was used to determine which tests to run on AWS
+// As a contrived example, let's say we want to run these same tests on GovCloud
+awsTests := ginkgoExtensionSpecs.Select(extensions.PlatformSelector(platforms.AWS))
+awsTests.AddPlatform(platforms.GovCloud)
+
+// And so on with expressive grouping & metadata management. awsTests
+// references the same objects as ginkgoExtensionSpecs, so changes in
+// one are reflected immediately in both sets.
+awsTests.Select(extensions.LabelSelector(platform.SLOW)).SetResource(resource.CPU, "5000m")
+
+// Since this set of tests doesn't use the OpenShift naming conventions, they
+// must constrain environmental parameters themselves.
+customExtensionSpecs.Select(extensions.NameRegExSelector(".*cld/aws.*")).SetPlatforms(platforms.AWS)
+
+// You can add hooks to execute BeforeAll, BeforeEach, AfterAll, AfterEach for 
+// ExtensionTestSpecs.  These are sticky to that group individual specs, so you're still
+// able to use different initialization code per spec set.
+customExtensionSpecs.AddBeforeAll(func() {
+    // perform custom extension initialization
+})
+
+ginkgoExtensionSpecs.AddBeforeAll(func() {
+    // perform custom ginkgo initialization
+})
+
+// Since everything is an ExtensionTestSpec, you can mix sets.
+allSpecsSet := customExtensionSpecs
+allSpecsSet.Add(ginkgoExtensionSpecs)
+
+allSpecsSet.AddAfterAll(func() {
+    // perform clean up that applies to both sets
+})
+
+// The author can iterate through all tests and, for example,
+// set OriginalName for each based on a configuration file
+// or dictionary maintained in their repository.
+allSpecsSet.Walk(func(*ExtensionTestSpec) {
+    // do stuff
+})
+
+// Finally, some or all of the specs are added for the component. 
+componentExtension.AddSpecs(allSpecsSet...)
+
+customSuite := extension.Suite{
+    Name: "my-custom-suite",
+    Parents: []string{
+        "openshift/conformance/parallel",
+    },
+    Qualifiers: []string{
+        // CEL expression which selects appropriate tests
+        "(test.tags.mytag==\"myvalue\"", 
+    }
+}
+
+// AddGlobalSuite adds a suite whose qualifiers will extend to all test binaries
+componentExtension.AddGlobalSuite(customSuite)
+
+// AddSuite adds a suite whose qualifiers are automatically modified to filter on
+// source = this component.
+componentExtension.AddSuite(customSuite)
+
+// The first component in the NewExtensionRegistry argument list
+// will be selected by --component default, but also
+// selectable with --component openshift:payload:hyperkube . 
+// Most authors will be writing tests for one component, so origin
+// omitting an option for --component, unless instructed otherwise, will "just 
+// work".
+componentRegistry := e.NewRegistry()
+componentRegistry.Register(componentRegistry)

```

Comment:
@jupierce is it `componentRegistry.Register(componentExtension)`? thanks

Reply:
It is! Nice catch.

I'll correct this, but for implementers, better information on the state of the API can be found here:
https://docs.google.com/document/d/1cFZj9QdzW8hbHc3H0Nce-2xrJMtpDJrwAse9H7hLiWk/edit?tab=t.0#heading=h.66y4kqbj468a is the source for details on the API. https://github.com/openshift-eng/openshift-tests-extension/blob/main/cmd/example-tests/main.go provides an example.

Reply:
Thank you for the information!